modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageManagerImpl.java

```diff
@@ -889,26 +1027,45 @@ public CompletableFuture<IndexWithTerm> raftNodeIndex() {
 
     @Override
     public CompletableFuture<Void> becomeLonelyLeader(long termBeforeChange, Set<String> targetVotingSet) {
-        // TODO: IGNITE-22899 - use both parameters.
         return inBusyLockAsync(busyLock, () -> {
-            synchronized (becomeLonelyLeaderMutex) {
-                leaderSecondaryDutiesPaused = targetVotingSet.size() > 1;
+            synchronized (peersChangeMutex) {
+                if (peersChangeState != null) {
+                    return failedFuture(new IgniteInternalException(
+                            INTERNAL_ERR,
+                            "Peers change is under way [state=" + peersChangeState + "]."
+                    ));
+                }
+
+                peersChangeState = targetVotingSet.size() > 1 ? new PeersChangeState(termBeforeChange, targetVotingSet) : null;
```

Comment:
didnt' grasp, why it's null if targetVotingSize <= 1?

Reply:
I added a comment which explains this